R14 - [Cliente] Cadastro de Cartões

FilipeMSouza opened the pull request:
[card](https://trello.com/c/9m0lEu94/13-r14-cliente-gerenciar-cart%C3%B5es-de-cr%C3%A9dito)
- Cadastro de cartões
- Grupoi revisor Grupo 2
- entrega para revisao [6/12/2021]

N0tFake committed:
feature/R14/silvio/cadastro_cartão

- Adicionando front do gerenciador de cartões

FilipeMSouza committed:
feture/r15/t01/NP_form | Cadros novo produto

- Form para cadastro de novo produto
- Criação de dois componentes para otimização de codigo
- Alteração de widget para StatefullWidget no componente AddProduct

FilipeMSouza committed:
feature/r15/t01/NP_form | Remoção de controlers

- No commit anterior eu tinha criado alguns controlers que eram
inuteis, sendo assim eles foram removidos

FilipeMSouza committed:
Merge branch 'develop' of https://github.com/FilipeMSouza/proj-sist-2021-2 into feature/R15/t01/NP_form

N0tFake committed:
feature/R14/silvio/cadastro card

- Adicionado Dialog para cadastro do cartão
- Criado formulario de cadastro
- Atribuido as validações no formulario

FilipeMSouza committed:
feature/R15/t01/NP_form | Seleção de imagem

- Dependencia para selecionar imagem
- Seleção de imagem como forma de um botão

vhgomes committed:
feature/R15/t01/ - Adicionado repositories
- Adicionado os arquivos do repositories e esboço de função para adicionar produto

vhgomes committed:
feature/R15/t01/ - Erro de digitação consertado
- Pequeno erro de digitação consertado arrumado.

vhgomes committed:
feature/R15/t01/ - Valor no store adicionado
 - Valor de available adicionado para ser um observavel

vhgomes committed:
feature/R15/t01/ - Reformulado form
- Form de adicionar produto foi modificado para melhor leitura de codigo
- Agora o slider de disponivel e um observavel.

vhgomes committed:
feature/R15/t01/ - Adicionado rota
- Rota do HomeRestaurant adicionado no appmodule

vhgomes committed:
feature/R15/t02/ - Funções refomurladas
- Reescritas para serem async/await.

vhgomes committed:
feature/R15/t02/ - Função de adicionar funcional
- Função para adição de produto no banco está a funcionar
- Inserido controller no arquivo custom_text_field.dart
OBS: Os campos de imagem e categorias ainda em desenvolvimento.

vhgomes committed:
feature/R15/t02/ - Adicionado campos
- Adicionado campos no restaurant_home_store e a chamada da função cadastro
de produto no banco.

FilipeMSouza committed:
feature/R15/t01/NP_fomr | Selecionar imagem

- Um novo componente de criação de imagem para um modele de drag and
  drop

FilipeMSouza committed:
feature/R15/t01/NP_form | Selecionar imagem

- Modo de seleção de imagem alterado para conincidir com o modelo
  adotado pelo grupo 1, usando um drag and drop zone

vhgomes committed:
feature/R15/t02/ - Retirado imports sem uso

vhgomes committed:
feature/R15/t02/ - Validação simples adicionada
- Pequena validação dos campos foi adicionado

FilipeMSouza committed:
feature?r15/t01/NP_form | TextFormfield

- ajsute no campo de cadastro de preço para ser validado da forma
  correta

FilipeMSouza committed:
feature/R15/t01/NP_form | Ajuste na formatação

- A formatação do campo de preço foi alterada para condizer com outras
  partes do projeto

vhgomes committed:
Merge branch 'develop' of https://github.com/FilipeMSouza/proj-sist-2021-2 into feature/R15/t01/NP_form

vhgomes committed:
feature/R15/t012/ - Removidas funções sem uso.

vhgomes committed:
feature/R15/t02/ - Conflitos gerados pela merge
Esse commit foi feito para arrumar erros gerados pelo merge com a branch
principal com o trabalho que estava sendo feito.
- Realocação dos arquivos
- Adequação para padronização de codigo para a inserção dos dados do produto.
- Mudanças nas classes de field para melhor escrita de codigo.

FilipeMSouza committed:
feature/R15/T01/NP_form | Comunicação com Firebase

- Envia a img para a storage do firebase

FilipeMSouza committed:
feature/R15/t01/NP_NP_form | envio da imagem

- Imagem eviada para o firebaseStorage com o id do item
- Item cadastrado no banco

FilipeMSouza committed:
feature/R15/t01/NP_form | Ajustes finais

- Ultimos ajustes pré correção

FilipeMSouza committed:
feature/R15/t01/NP_form | atualização de categoria

- Categoria nao precisa ser um array então passamos ela para um string e
  atualizamos a passagem dela para o banco
- Removemos o Id do item pois nao tem necessidade dele ja que o
  documento do firebase ja tem um id proprio

FilipeMSouza committed:
develop | Remove de imports

- import inuteis removidos

FilipeMSouza committed:
develop | Correções pedidas

- Verificação de tipo de arquivo, agora podendo somente enviar jpg ou
  png

- O botão de selecionar da galeria estava bugando e travando a aplicação
  quando a verificação de arquivo era implementada, então deicidi tirar
  ele

FilipeMSouza committed:
develop | Resolução de Conflitos

- Conflitos que o git alertou foram resolvidos
- o unico codigo relevante para a correção são os de dentro da pasata
  restaurant_home na pasta components e add_product

FilipeMSouza committed:
develop | notificações de ações

- Alerta de imagens inserida e produto cadastrado
- Alerta de imagem não suportada atualizada

FilipeMSouza committed:
develop | Alterações do formulário

- O botão img foi trocado por um icone
- A posição do botão do foi colocada pra baixo

FilipeMSouza committed:
develop | Correções

- Reposicionamento e ajuste do dialog do usuario

FilipeMSouza committed:
develop | Revisão

- Correção de alguns bugs e posicionamentos dentro do dialog

vhgomes committed:
Revisão - R15 | Imports retirados

vhgomes committed:
Revisão - R15 - Modificado nome da função

vhgomes committed:
Revisão - R15 | Reformulação da função de envio para banco
- Retirado imports não utilizados
- O nome da função agora está correto
- Retirado toda a gambiarra criando uma classe no add_product_card e enviando pra função, agora ela está
dentro da store para melhor visualização do codigo.

vhgomes committed:
Revisão - R15 | Input de preço arrumado
- Retirado o toList() pois estava sendo convertido duas vezes para lista
- Inserido validação de preço.

vhgomes committed:
cadastrocartao| R14 - T02 - Conexão com o banco feita
- Adicionado a funcionalidade de adicionar cartão.
- Reformulado classe de cartão
- Retirado pasta de testes

N0tFake committed:
cadastrocartao| R14 - Frontend correções pontuais

vhgomes committed:
cadastrocartao| R14 - T02 - Correções
- Antes os cartoes cadastrados eram apagados ao inserir outro, agora ele adiciona na lista

FilipeMSouza committed:
feature/R14/silvio/cadastro_catao

- Problemas indicados pela IDE

FilipeMSouza committed:
develop | Correções pedidas

- Notificação de cadastro de produto no banco
- Campos limpos após fechar o dialog

FilipeMSouza committed:
Develop | Correção

- Correções pedidas pelo grupo 2 que foi o revisor

FilipeMSouza committed:
develop | Correção de um erro de cadastro

- Correção que cadastrava o produto quando o botão era acionado mesmo
  com o produto vazio

FilipeMSouza committed:
develop | correção

- Validação agora funciona de forma certa
	- Foi colocado uma formKey para validar os campos
	- Continuamos trabalhando com o store.imgPath agr funcinando de
	  forma correta
- Na função de inserção de image mudamos a posição da função
  setProductImage com a imageAlert para facilitar a verificação na hora
  de salvar o produto

FilipeMSouza committed:
develop | Correção

- Verificação de imagem, antes a varavel imgPath ficava sempre com o
  valor anterior, agora quando ela é cadastrada ela assume um valor nulo
  novamente e não consegue cadastrar o produto

FilipeMSouza committed:
hotfix/Auxilar_firebase | erro de cuminicação (#12)

- No momento de limapar os campos do form a imagem que iria pro banco de
  dados erra passada como null e nao enviava para o banco

FilipeMSouza committed:
Merge branch 'develop' into develop

FilipeMSouza committed:
hotfix/integracao_dev+R14 | integração

- A feature de adicionar cartao foi integrada a develop

FilipeMSouza committed:
develop | merge com a branch de integracao

- um arquivo que nao tinha sido commitado pubspec.lock

FilipeMSouza committed:
Sincronização com a develop principal

N0tFake committed:
teste

N0tFake committed:
teste

djesusguimaraes commented on lib/app/global/repositories/paymentcard/payment_card.dart:
```diff
@@ -0,0 +1,41 @@
+        'cvv': card.cvv,
+        'name': card.nome,
+        'validade': card.validity,
+      }).then((value) {
```
Não falta o `CPF` do dono do cartão?

vhgomes replied:
O CPF é atrelado a conta do usuário logado, acredito que não seja necessário colocar no registro do cartão. Seria meio que um dado duplicado dentro da nossa tabela de usuario.

vhgomes replied:
![image](https://user-images.githubusercontent.com/60026761/145662861-6f82b4d9-ccf5-40e2-acba-3a73dcf78937.png)

![image](https://user-images.githubusercontent.com/60026761/145662869-e3094be4-71ca-4e85-b1cf-4178a6490afd.png)

O problema pode ser o seguinte, caso ter sido cadastrado por  meio de terceiros (Google, Facebook) pois ai não terá o campo de CPF, porém caso for registrado pelos campos normalmente, o CPF já vai estar cadastrado na conta.

djesusguimaraes replied:
Não há garantia que o cartão é do cliente logado. Por isso o cadastro de cartão em todas as plataformas digitais pede o CPF vinculado ao cartão, sem levar em conta a relação com a conta de usuário no site/plataforma.

Ex, próprio Ifood:
![image](https://user-images.githubusercontent.com/66970708/145681443-87220e0d-9c3c-4c28-9651-fafdab079b2c.png)

vhgomes replied:
Saquei saquei, mas como eu sou rápido isso ai já foi arrumado, inseri o campo aqui. Valeu.

djesusguimaraes commented on lib/app/modules/payments/components/add_card/TextField/text_field_card.dart:
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/material.dart';
```
**Sobre os arquivos anteriores**
* As imagens e assets não deveriam estar na pasta global do projeto?
  `proj-sist-2021-2/assets`

djesusguimaraes commented on lib/app/modules/payments/components/add_card/add_body.dart:
```diff
@@ -0,0 +1,60 @@
+        Container(
+          color: primaryCollor,
+          width: 300,
+          height: 50,
```
**Observers desnecessários**
* Dada a estrutura de widgets o `Material` na linha `29` não necessita ser retornado por um Observer, já que o foco final é o `IconButton`, que já é retornado por um Observer.

N0tFake committed:
RW03/T04/silvio/view_card | motrar cartões

- Cartões cadastrados no BD, agora podem ser visualizados no site
- Mudança de rota ao adicionar um cartão. Agora o usuario é
  redirecionado para a home

djesusguimaraes commented on lib/app/modules/payments/components/add_card/form_card.dart:
```diff
@@ -0,0 +1,177 @@
+                    return SizedBox(
+                        width: 100,
+                        child: textFormCard(
+                            hintText: 'Numero do cartão',
```
**Falta de limite e máscara**

O campo aceita mais que os caracteres necessários.
Além de não organizar a entrada dos números.

Esse tipo de comportamento pode dificultar a experiência do usuário final.

djesusguimaraes replied:
Todos os campos aceitam letras (String) como entrada, mesmo para valores declaradamente numéricos.

N0tFake committed:
RW03/T04/silvio/view_card | correção 01

- Correção na cor do CircularProgressIndicator
- Correção na duplicata de cartões

djesusguimaraes commented on lib/app/modules/payments/components/cards_payments/list_card.dart:
```diff
@@ -0,0 +1,80 @@
+
+    return Observer(builder: (_) {
+      return ListView.builder(
+          itemCount: 1, // numero de cartões cadastrados
```
**Dados Estáticos**

Os dados da listagem de cartão estão estáticos, logo se explica a falta de atualização ao cadastrar um novo cartão.

djesusguimaraes reviewed (changes requested):
## Detalhes

Como houve impedimento nos testes causado por dados estáticos considero REPROVADO.

djesusguimaraes commented:
## Observação

**Lembrem-se de atualizar a branch para resolver os conflitos**

vhgomes committed:
feature/R14/silvio/cadastro_cartao - CPF Inserido
- Inserido campo de CPF na classe.
- Inserido CPF na inserção no banco de dados.

vhgomes committed:
Merge branch 'feature/R14/silvio/cadastro_cartao' into develop

FilipeMSouza committed:
feature/R14/silvio/cadastro_catrao | Correção

- Pedidos da correção

FilipeMSouza committed:
Correção de conflitos

FilipeMSouza committed:
develop | Atualização

- Atualizamos a develop com o upstream develop

FilipeMSouza committed:
Correção de um bug na pubspec.yaml

N0tFake committed:
RW03/T04/silvio/view_card | correção part 2

- Corrigdido erro quando o cliente não possuisse cartão cadastrado

N0tFake committed:
RW03/T04/silvio/view_card | correção part 2

- Corrigdido erro quando o cliente não possuisse cartão cadastrado

N0tFake committed:
RW03/T04/silvio/view_card | correção part 2

- Corrigido error quando o cliente não possuisse cartão cadastrado.

vhgomes committed:
Merge branch 'RW03/T04/silvio/view_card' into develop

vhgomes committed:
develop| BUGFIX - Erro ao puxar CPF consertado
- Antes por causa de conflito, o CPF não estava sendo puxado do banco, com esse commit ele volta a funcionar normalmente.

djesusguimaraes commented:
## Conclusão

Dada a semântica do requisito, que trata apenas da inserção de cartões, considero aceitável a falta das funções de deleção e atualização.

Considero APROVADO.
